Weather App/controller/ViewController.swift

```diff
@@ -0,0 +1,66 @@
+import CoreLocation
+import UIKit
+
+class ViewController: UIViewController, UITableViewDataSource {
+    @IBOutlet var weatherTableView: UITableView!
+    private let userLocationViewModel = UserLocationViewModel()
+    var weatherForCityViewModel: WeatherForCityViewModel = .init()
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Do any additional setup after loading the view.
+        userLocationViewModel.configureLocationService()
+
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(coordinateChanged),
+                                               name: .coordinateChanged,
+                                               object: nil)
+
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(receivedWeather(_:)),
+                                               name: .weatherReturned,
+                                               object: nil)
+        weatherTableView.register(UINib(nibName: "WeatherCell", bundle: nil), forCellReuseIdentifier: "WeatherCell")
+        weatherTableView.dataSource = self
+        weatherTableView.backgroundColor = .white
+    }
+
+    @objc func coordinateChanged(_: Notification) {
+        guard let coordinate = userLocationViewModel.userLocation.currentLocation else {
+            return
+        }
+        weatherForCityViewModel.getWeatherForCoordinate(coordinate: coordinate)
+    }
+
+    @objc func receivedWeather(_ notification: Notification) {
+        guard let weather = notification.object as? WeatherForCity else {
+            return
+        }
+        print(weather.mesurements.temperature)
+
+        DispatchQueue.main.async {
+            self.weatherTableView.reloadData()
+        }
+    }
+
+    func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {
```

Comment:
it's nice to keep table view delegates below of classes in extension. It is easier to read.

`extension ViewController: UITableViewDataSource {

func tableView(_: UITableView, numberOfRowsInSection _: Int) -> Int {}

func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {}

}`